src/filebackedarray/Hdf5DelayedArray.py

```diff
@@ -0,0 +1,59 @@
+from delayedarray import DelayedArray
+
+from .utils import H5DatasetInfo
+
+__author__ = "jkanche"
+__copyright__ = "jkanche"
+__license__ = "MIT"
+
+
+class Hdf5DelayedArray(DelayedArray):
+    """Base class for H5 backed arrays.
+
+    May be used to identify a `delayed` object with ``isinstance``.
+    """
+
```

Comment:
You don't need `shape` or `dtype`, this should be obtained from `seed` if you pass the seed to the parent's constructor.